exareme2/controller/services/flower/flower_algorithm_registry.py

```diff
@@ -0,0 +1,68 @@
+import asyncio
+from enum import Enum
+from enum import unique
+from logging import Logger
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from pydantic import BaseModel
+
+
+@unique
+class Status(str, Enum):
+    SUCCESS = "success"
+    RUNNING = "running"
+    FAILURE = "failure"
+
+
+class AlgorithmInputDataDTO(BaseModel):
+    data_model: str
+    datasets: List[str]
+    filters: Optional[dict]
+    y: Optional[List[str]]
+    x: Optional[List[str]]
+
+
+class FlowerAlgorithm(BaseModel):
+    inputdata: AlgorithmInputDataDTO = AlgorithmInputDataDTO(data_model="", datasets=[])
+    result: Dict = {}
+    status: Status = Status.SUCCESS
+
+
+class FlowerAlgorithmRegistry:
+    def __init__(self, logger):
+        self._logger = logger
+        self._flower_algorithm = FlowerAlgorithm()
+        self.lock = asyncio.Lock()
+        self.result_ready = asyncio.Event()
```

Comment:
If we have the event, why do we need the lock?

Reply:
We will need them once we have concurrent execution which is of the table for now.